whisper.py

```diff
@@ -920,12 +921,25 @@ def file_fetch(fh, fromTime, untilTime, now=None):
     untilTime = now
 
   diff = now - fromTime
+
+  #Parse granularity if requested
+  if archiveToSelect:
+    retentionStr = str(archiveToSelect)+":1"
+    archiveToSelect = parseRetentionDef(retentionStr)[0]
+
   for archive in header['archives']:
-    if archive['retention'] >= diff:
-      break
+    if archiveToSelect:
+      if archive['secondsPerPoint'] == archiveToSelect:
+        break
+      archive = None
+    else:
+      if archive['retention'] >= diff:
+        break
 
-  return __archive_fetch(fh, archive, fromTime, untilTime)
+  if archiveToSelect and not archive:
+    raise ValueError("Invalid granularity: %s" %(archiveToSelect))
 
+  return __archive_fetch(fh, archive, fromTime, untilTime)
 
```

Comment:
Please replace the blank line removed here, then I think this will be good to merge